extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/grpc/GrpcTracingServerInterceptor.java

```diff
@@ -99,65 +98,87 @@ public String get(final GrpcRequest carrier, final String key) {
 
     private class TracingServerCallListener<ReqT> extends SimpleForwardingServerCallListener<ReqT> {
         private final Context spanContext;
+        private final Scope scope;
         private final GrpcRequest grpcRequest;
 
-        protected TracingServerCallListener(final ServerCall.Listener<ReqT> delegate, final Context spanContext,
+        protected TracingServerCallListener(
+                final ServerCall.Listener<ReqT> delegate,
+                final Context spanContext,
+                final Scope scope,
                 final GrpcRequest grpcRequest) {
+
             super(delegate);
+            this.scope = scope;
             this.spanContext = spanContext;
             this.grpcRequest = grpcRequest;
         }
 
         @Override
         public void onHalfClose() {
-            try (Scope ignored = spanContext.makeCurrent()) {
+            try {
                 super.onHalfClose();
             } catch (Exception e) {
-                instrumenter.end(spanContext, grpcRequest, null, e);
+                try (scope) {
```

Comment:
this is just to close the scope on exit right?

Reply:
Yes, we need to close to scope on every possible exit.